app/bundles/EmailBundle/Views/FormTheme/Config/_config_emailconfig_widget.html.twig

```diff
@@ -49,37 +58,24 @@
                 </div>
             {% endif %}
 
-            <div class="row">
-                {{ formRowIfExists(fields, 'mailer_amazon_region', template) }}
-                {{ formRowIfExists(fields, 'mailer_amazon_other_region', template) }}
-            </div>
-
-            <div class="row">
-                {{ formRowIfExists(fields, 'mailer_sparkpost_region', template) }}
-            </div>
-
             <div class="row">
                 {{ formRowIfExists(fields, 'mailer_host', template) }}
                 {{ formRowIfExists(fields, 'mailer_port', template) }}
             </div>
 
-            <div class="row">
-                {{ formRowIfExists(fields, 'mailer_encryption', template) }}
-                {{ formRowIfExists(fields, 'mailer_auth_mode', template) }}
-            </div>
-
             <div class="row">
                 {{ formRowIfExists(fields, 'mailer_user', template) }}
                 {{ formRowIfExists(fields, 'mailer_password', template) }}
                 {{ formRowIfExists(fields, 'mailer_api_key', template) }}
             </div>
 
-            {% if fields.mailer_transport is defined %}
-                <div class="row">
-                    {{ formRowIfExists(fields, 'mailer_mailjet_sandbox', template) }}
-                    {{ formRowIfExists(fields, 'mailer_mailjet_sandbox_default_mail', template) }}
-                </div>
-            {% endif %}
+            <div class="row">
+                {% for key in fieldKeys|filter(key => key matches '/^mailer_option/') %}
```

Comment:
@RCheesley @RCheesley if we have complex `Dsn` settings, I suggest we remove the ability to change email settings from the UI and force admins to change from the config

Reply:
How do we define complex here? Thinking from the user experience that would be a really clunky experience, I think. It should be one or the other for all, and my preference would be through the UI.

Reply:
I would put the DSN field in the email configuration instead of all the separate fields we have there now. And write a good explanation of what each piece of the DSN string means bellow it so people could build it themselves. I think many people would be fine with that. And they could simply copy-paste the examples and fill in their values in.

Reply:
Vote for @escopecz suggestions

Reply:
The email configuration UI has been revamped. It allows to set all the parts of the DNS now.